src/Resources/contao/library/Contao/StringUtil.php

```diff
@@ -547,13 +558,27 @@ public static function parseSimpleTokens($strString, $arrData)
 
 		// Replace tokens
 		$strReturn = str_replace('?><br />', '?>', $strReturn);
-		$strReturn = preg_replace('/##([A-Za-z0-9_]+)##/i', '<?= $arrData[\'$1\'] ?>', $strReturn);
-		$strReturn = str_replace("] ?>\n", '] . "\n" ?>' . "\n", $strReturn); // see #7178
+		$strReturn = preg_replace_callback(
+			'/##([^=!<>\s]+)##/',
+			function (array $matches) use ($arrData)
+			{
+				if (!array_key_exists($matches[1], $arrData))
+				{
+					System::log(sprintf('Tried to parse unknown simple token "%s".', $matches[1]), __METHOD__, TL_ERROR);
+					return '##' . $matches[1] . '##';
+				}
+
+				return $arrData[$matches[1]];
+			},
+			$strReturn
+		);
+		$strReturn = str_replace("]; ?>\n", '] . "\n"; ?>' . "\n", $strReturn); // see #7178
 
 		// Eval the code
 		ob_start();
 		$blnEval = eval("?>" . $strReturn);
-		$strReturn = ob_get_clean();
```

Comment:
Why did you change this? `ob_get_clean()` is the same as `ob_get_contents()` + `ob_end_clean()`.